src/components/FeatureSetsPanel/FeatureSetsPanelDataSource/featureSetsPanelDataSource.util.js

```diff
@@ -1,4 +1,85 @@
+import { MLRUN_STORAGE_INPUT_PATH_SCHEME } from '../../../constants'
+
 export const kindOptions = [
   { label: 'CSV', id: 'csv' },
   { label: 'PARQUET', id: 'parquet' }
 ]
+
+export const comboboxSelectList = [
+  {
+    className: 'path-type-store',
+    label: 'MLRun store',
+    id: 'store://'
+  },
+  {
+    className: 'path-type-v3io',
+    label: 'V3IO',
+    id: 'v3io://'
+  },
+  {
+    className: 'path-type-s3',
+    label: 'S3',
+    id: 's3://'
+  },
+  {
+    className: 'path-type-az',
+    label: 'Azure storage',
+    id: 'az://'
+  }
+]
+
+export const generateComboboxMatchesList = (
+  url,
+  artifacts,
+  projects,
+  projectName,
+  urlProjectPathEntered,
+  urlArtifactPathEntered,
+  urlArtifactReferencePathEntered,
+  artifactsReferences,
+  urlProjectItemTypeEntered
+) => {
+  if (!urlProjectItemTypeEntered) {
+    if (
+      projectItemsPathTypes.find(type =>
+        type.id.startsWith(url.projectItemType)
+      )
+    ) {
+      return projectItemsPathTypes
+    } else {
+      return []
+    }
+  } else if (!urlProjectPathEntered) {
+    return projects.filter(project => {
+      return project.id.startsWith(url.project)
+    })
+  } else if (!urlArtifactPathEntered) {
+    return (artifacts ?? []).filter(artifact => {
+      return artifact.id.startsWith(url.artifact)
+    })
+  } else if (!urlArtifactReferencePathEntered) {
+    return (artifactsReferences ?? []).filter(projectItem => {
+      return projectItem.id.startsWith(url.artifactReference)
+    })
+  } else {
+    return []
+  }
+}
```

Comment:
To clean this entire `if-else if-else if-else` structure, you can skip the `else`s, since each `if` branch ends with a `return`.

Consider:
```js
if (condition) {
  // statements if condition is true
  return valueIfTrue;
} else {
  // statements if condition is false
  return valueIfFalse;
}
```
You could do this instead:
```js
if (condition) {
  // statements if condition is true
  return valueIfTrue;
}
// statements if condition is false
return valueIfFalse;
}
```
This is called “implicit else”.